vtex-apps/phone-field
# Reuse country-flags and checkout-components apps

#### What problem is this solving?

This PR updates this component to use the newly created `vtex.country-flags` app and the `Listbox` component from `vtex.checkout-components`. I updated the listbox border styling to be synced with the input, and changed the mask to only be applied on the `blur` event.

I also fixed a bug when the rules array was empty the component would crash.

#### How should this be manually tested?

[Workspace](https://steps--checkoutio.myvtex.com/checkout/#/profile). You can see the updated component in the profile step.

#### Checklist/Reminders


#### Screenshots or example usage

| Before | After | Comment |
| --- | --- | --- |
| ![phone-field-old-1](https://user-images.githubusercontent.com/10223856/84314402-5fb15280-ab3e-11ea-9257-24088b2540c9.gif) | ![phone-field-new-1](https://user-images.githubusercontent.com/10223856/84314430-6a6be780-ab3e-11ea-9348-dae496edbbb2.gif) | Border color and mask on blur |
| ![image](https://user-images.githubusercontent.com/10223856/84314531-9d15e000-ab3e-11ea-999b-7b42b2ffb627.png) | ![image](https://user-images.githubusercontent.com/10223856/84314559-aacb6580-ab3e-11ea-85c7-b2821f1a102f.png) | Order of countries |

#### Type of changes


✔️ | Type of Change
---|---
✔️ | Bug fix 
_ | New feature 
_ | Breaking change 
_ | Technical improvements
